File: servers/home/GridPathfinder.js
/** @param {NS} ns */
export async function main(ns) {

// Computes the number of unique paths from the top-left corner
// to the bottom-right corner of an r×c grid, moving only down or right.

// {number} r - Number of rows in the grid.
// {number} c - Number of columns in the grid.
// Returns {number} The total number of unique paths.

  function numberOfPaths(r, c) {
    // Base case: if there's only one row or one column,
    // there is exactly one path (all the way right or all the way down).
    if (r === 1 || c === 1) {
      return 1;
    }

    // Recursive step: sum of paths from the cell above (r-1, c)
    // and from the cell to the left (r, c-1)
    return numberOfPaths(r - 1, c) + numberOfPaths(r, c - 1);
  }

  // Plug in the numbers of the grid you need:
  const r = 8;
  const c = 6;

  ns.tprint(numberOfPaths(r, c));  // Outputs the number of paths for a grid.
}

File: servers/home/utils/GridPathfinder.py
#You are in a grid with 8 rows and 6 columns, and you are positioned in the top-left corner of that grid. You are trying to reach the bottom-right corner of the grid, but you can only move down or right on each step. Determine how many unique paths there are from start to finish.
#Data should be displayed as an array, i.e. [8, 6]

def numberOfPaths(r, c):
  if r == 1 or c == 1:
    return 1

  return numberOfPaths(r-1, c) + numberOfPaths(r, c-1)

    r: int = 8
    c: int = 6

print(numberOfPaths(r, c))
